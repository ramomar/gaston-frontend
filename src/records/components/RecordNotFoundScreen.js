import React from 'react';
import PropTypes from 'prop-types';
import { LinkPrevious } from 'grommet-icons';
import { Box, Text, Button } from 'grommet';
import { Screen, ScreenHeader, ScreenBody } from '../../foundation/components/screen';

function RecordNotFoundScreen(props) {
  return (
    <Screen>
      <ScreenHeader
        start={<Button plain icon={<LinkPrevious />} onClick={props.goToRecords} />}
        center={<Text weight='bold' size='large'>Gasto no encontrado</Text>} />
      <ScreenBody>
        <Box fill='vertical' justify='center'>
          <Text textAlign='center'>¡Ese gasto no se encontró!</Text>
        </Box>
      </ScreenBody>
    </Screen>
  );
}

RecordNotFoundScreen.propTypes = {
  goToRecords: PropTypes.func.isRequired
};

export default RecordNotFoundScreen;
