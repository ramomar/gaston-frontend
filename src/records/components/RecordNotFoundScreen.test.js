import React from 'react';
import renderer from 'react-test-renderer';
import { Grommet } from 'grommet';
import RecordNotFoundScreen from './RecordNotFoundScreen';

it('renders correctly', () => {
  const component =
    <Grommet>
      <RecordNotFoundScreen
        goToRecords={() => null}>
      </RecordNotFoundScreen>
    </Grommet>;

  const tree = renderer.create(component);

  expect(tree).toMatchSnapshot();
});
