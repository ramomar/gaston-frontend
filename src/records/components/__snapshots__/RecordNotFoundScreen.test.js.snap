// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <div
    className="StyledBox-sc-13pk1d4-0 tkzRw"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 fFWPAz"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 fWrrBh"
      >
        <button
          className="StyledButton-sc-323bzc-0 laBEqz"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <svg
            aria-label="LinkPrevious"
            className="StyledIcon-ofa7kd-0 iOkQrb"
            viewBox="0 0 24 24"
          >
            <path
              d="M2,12 L22,12 M13,3 L22,12 L13,21"
              fill="none"
              stroke="#000"
              strokeWidth="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </button>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 fWrrBh"
      >
        <span
          className="StyledText-sc-1sadyjn-0 dFefMJ"
          size="large"
        >
          Gasto no encontrado
        </span>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 fWrrBh"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 tkzRw"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 eQUjUa"
      >
        <span
          className="StyledText-sc-1sadyjn-0 dmvzzC"
        >
          ¡Ese gasto no se encontró!
        </span>
      </div>
    </div>
  </div>
</div>
`;
